crates/sui-rest-api/src/checkpoints.rs

```diff
@@ -100,13 +106,24 @@ pub async fn get_full_checkpoint(
             )
             .collect::<HashSet<_>>()
             .into_iter()
+            // Unwrapped-then-deleted objects are not stored in state before the tx, so we have nothing to fetch.
```

Comment:
This is not an issue in this PR (so not blocking), but just noting that the pattern of collecting into a `HashSet` and then draining that into a `Vec` makes me nervous from a determinism perspective -- i.e. the response order could change non-deterministically -- is there a reason we can't use a `BTreeSet` instead here?

Reply:
that's a valid point but i don't think the order matters here

Reply:
I agree, order doesn't matter, but what I'm more concerned about is that having a stable order can be important to make testing easier, etc.

Reply:
that's actually a good point re testability.